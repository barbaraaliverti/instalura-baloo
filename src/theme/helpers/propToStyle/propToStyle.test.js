import propToStyle from './index';

describe('propToStyle()', () => {
  describe('when a simple argument is received', () => {
    test('and it is a string', () => {
      const propToStyleResult = propToStyle('textAlign');
      const componentProps = { textAlign: 'center' }; // string
      const styleResult = propToStyleResult(componentProps);
      expect(styleResult).toEqual({ textAlign: 'center' });
    });

    test('and it is a number', () => {
      const propToStyleResult = propToStyle('flex');
      const componentProps = { flex: 1 }; // number
      const styleResult = propToStyleResult(componentProps);
      expect(styleResult).toEqual({ flex: 1 });
    });
  });

  describe('when an argument with breakpoints is received', () => {
    test('renders only one breakpoint resolution', () => {
      const propToStyleResult = propToStyle('textAlign');
      const componentProps = { textAlign: { xs: 'center'} };
      const styleResult = propToStyleResult(componentProps);
      expect(styleResult).toMatchSnapshot();
    });

    test('renders two or more breakpoint resolutions', () => {
        const propToStyleResult = propToStyle('textAlign');
        const componentProps = { textAlign: { xs: 'center', md: 'right' } };
        const styleResult = propToStyleResult(componentProps);
        expect(styleResult).toMatchSnapshot();
      });
  });
});
