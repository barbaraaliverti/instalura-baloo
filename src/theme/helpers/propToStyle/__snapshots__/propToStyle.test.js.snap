// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`propToStyle() when an argument with breakpoints is received renders only one breakpoint resolution 1`] = `
Array [
  "@media screen and (min-width:",
  "0",
  "px){",
  "text-align: center;",
  "}",
  "     ",
]
`;

exports[`propToStyle() when an argument with breakpoints is received renders two or more breakpoint resolutions 1`] = `
Array [
  "@media screen and (min-width:",
  "0",
  "px){",
  "text-align: center;",
  "}",
  "@media screen and (min-width:",
  "768",
  "px){",
  "text-align: right;",
  "}",
  "     ",
]
`;
